src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/external/UserSearchControllerIntTest.kt

```diff
@@ -110,15 +101,121 @@ class UserSearchControllerIntTest : IntegrationTestBase() {
+        .jsonPath("$.pageable.unpaged").isEqualTo(false)
+        .jsonPath("$.pageable.sort.empty").isEqualTo(false)
+        .jsonPath("$.pageable.sort.sorted").isEqualTo(true)
+        .jsonPath("$.pageable.sort.unsorted").isEqualTo(false)
+        .jsonPath("$.sort.empty").isEqualTo(false)
+        .jsonPath("$.sort.sorted").isEqualTo(true)
+        .jsonPath("$.sort.unsorted").isEqualTo(false)
+    }
+
+    @Test
+    fun `should respond with paged results when users found`() {
+      val name = "tester.mctesty@digital.justice.gov.uk"
+      val roles = listOf("TESTING", "MORE_TESTING")
+      val groups = listOf("TESTING_GROUP", "MORE_TESTING_GROUP")
+      val rolesJoined = roles.joinToString(",")
+      val groupsJoined = groups.joinToString(",")
+
+      externalUsersApiMockServer.stubUserSearchAllFiltersWithResults(name, roles, groups)
+
+      webTestClient.get().uri("/externalusers/search?name=$name&roles=$rolesJoined&groups=$groupsJoined")
+        .headers(setAuthorisation(roles = listOf("ROLE_MAINTAIN_OAUTH_USERS")))
+        .exchange()
+        .expectStatus().isOk
+        .expectBody()
+        .jsonPath("$.content[*].userId").value<List<String>> { assertThat(it).hasSize(2) }
+        .jsonPath("$.content[0].userId").isEqualTo("006a9299-ef3d-4990-8604-13cefac706b5")
+        .jsonPath("$.content[0].username").isEqualTo("TESTER.MCTESTY@DIGITAL.JUSTICE.GOV.UK")
+        .jsonPath("$.content[0].email").isEqualTo("tester.mctesty@digital.justice.gov.uk")
+        .jsonPath("$.content[0].firstName").isEqualTo("Tester1")
+        .jsonPath("$.content[0].lastName").isEqualTo("McTester1")
+        .jsonPath("$.content[0].locked").isEqualTo(false)
+        .jsonPath("$.content[0].enabled").isEqualTo(true)
+        .jsonPath("$.content[0].verified").isEqualTo(true)
+        .jsonPath("$.content[0].lastLoggedIn").isEqualTo("2022-12-14T10:23:04.915132")
+        .jsonPath("$.content[0].inactiveReason").isEmpty
+        .jsonPath("$.totalPages").isEqualTo(19)
+        .jsonPath("$.totalElements").isEqualTo(185)
+        .jsonPath("$.last").isEqualTo(false)
+        .jsonPath("$.size").isEqualTo(10)
+        .jsonPath("$.number").isEqualTo(0)
+        .jsonPath("$.first").isEqualTo(true)
+        .jsonPath("$.numberOfElements").isEqualTo(2)
+        .jsonPath("$.empty").isEqualTo(false)
+        .jsonPath("$.pageable.offset").isEqualTo(0)
+        .jsonPath("$.pageable.pageSize").isEqualTo(10)
+        .jsonPath("$.pageable.pageNumber").isEqualTo(0)
+        .jsonPath("$.pageable.paged").isEqualTo(true)
+        .jsonPath("$.pageable.unpaged").isEqualTo(false)
+        .jsonPath("$.pageable.sort.empty").isEqualTo(false)
+        .jsonPath("$.pageable.sort.sorted").isEqualTo(true)
+        .jsonPath("$.pageable.sort.unsorted").isEqualTo(false)
+        .jsonPath("$.sort.empty").isEqualTo(false)
+        .jsonPath("$.sort.sorted").isEqualTo(true)
+        .jsonPath("$.sort.unsorted").isEqualTo(false)
+    }
```

Comment:
might have been easier to just create a json file ?

Reply:
Considered it, & can change if you feel strongly. This way is my preference as it is all in one file